### PR TITLE
Fixes MySQL migrations and brings them up to speed.

### LINKED ISSUE #8596: MySQL Databases.
I'm filled dbconfig.txt with my server details, uncommented ENABLE_STAT_TRACKING and SQL_ENABLED, and commented BAN_LEGACY_SYSTEM. Server says, that conenction to both (tgstation and feedback) databases established, but temporary jobbans doesn't working, it saves only till reboot. And when i entering the server it says: "ADMIN LOG: fireflashie/(FireFlashie) is a relatively new player, may consider watching them. AGE =   First seen = "
Migration controller created tables.
Sorry for bad English.

### PR DESCRIPTION
**Pull request by PJB3005**
Fixes #8596

This essentially reverts #7891 too, because it broke `log_client_to_db()`, but nobody ever realized because it never hit the server due to a secret repo conflict.

By the way the Secret repo conflict is purely copying the older code from the secret repo, so it's not _actually_ conflicting.

### CODE REVIEW

File: code/modules/migrations/mysql/mysql_controller.dm [hunk 1]
```diff
@@ -31,7 +31,10 @@ CREATE TABLE IF NOT EXISTS [TABLE_NAME] (
 
 	var/list/rows=list()
 	while(query.NextRow())
-		rows += list(query.item)
```
Comment (PJB3005) [inline code/modules/migrations/mysql/mysql_controller.dm:34]: THIS was the bad line that was causing the MySQL migrations to runtime and as such not work. I have no idea why the fuck this doesn't work but my method does work so whatever.

File: code/modules/migrations/mysql/mysql_controller.dm [hunk 2]
```diff
@@ -31,7 +31,10 @@ CREATE TABLE IF NOT EXISTS [TABLE_NAME] (
...
+		world.log << json_encode(query.item)
+		rows[++rows.len] = query.item.Copy()
+
+	world.log << json_encode(rows)
```
Comment (clusterfack) [inline code/modules/migrations/mysql/mysql_controller.dm]: watafak

**Comment by gbasood** [comment]
By the way, we get `Error #0` outputted to admins and probably debug log when a new player joins
Error code #0 isn't an error though it's literally the equivalent of an OK response

So if you wanna fix that getting output that'd be neato

**Comment by PJB3005** [comment]
Isn't that outputted by SQLite, though?

**Comment by gbasood** [comment]
I mean maybe
If that's the case then it'd be _unatomic_ so nevermind

**Comment by StarSmasher44** [comment]
Tried running it on my own code (fully updated with /vg/),
I can't import the players2.sql file either way, error showing: `#1170 - BLOB/TEXT column 'player_ckey' used in key specification without a key length`

Also a crash on startup (with SQL enabled) but I'm figuring out of that is because of this or an unrelated issue.

**Comment by PJB3005** [comment]
You're not supposed to load `players2.sql` into MySQL, it's for the SQLite file in the repo root as reference. Copy `players2_empty.sqlite` to `players2.sqlite` for that.

Server crash means that you're not correctly connecting to the SQL server, it seems like BYOND's MySQL implementation causes the DLL to throw an exception if not connected. Check the server log (scroll up a bit) and it should report errors for who it isn't connecting.

Also just a tip: make sure the feedback database and the tgstation database are the same database because else migrations will fuck up.

**Comment by StarSmasher44** [comment]
Yeah I managed to get all of it to work, but there should be something making sure the server doesn't crash and trip out.
(Mine began displaying hundreds of runtimes on all config.variables so i'm gonna have to figure out how I managed to do that)

**Pull request merged**